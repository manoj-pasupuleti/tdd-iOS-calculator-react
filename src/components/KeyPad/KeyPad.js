import React from "react";
import PropTypes from "prop-types";

import "./KeyPad.css";
import Key from "../Key/Key";

export default function KeyPad({
  handleKeyDown,
  keyPadOperators,
  keyPadNumbers,
  updateDisplay,
  setOperator,
  callOperator
}) {
  const renderNumbers = () =>
    keyPadNumbers.map(number => (
      <Key
        handleKeyDown={handleKeyDown}
        key={`o_${number}`}
        keyValue={number}
        keyType="number-key"
        keyAction={updateDisplay}
      />
    ));

  const renderOperators = () =>
    keyPadOperators.map(operator => (
      <Key
        handleKeyDown={handleKeyDown}
        key={`o_${operator}`}
        keyValue={operator}
        keyType="operator-key"
        keyAction={setOperator}
      />
    ));

  return (
    <div className="keypad-container">
      <div className="numbers-container">{renderNumbers()}</div>
      <div className="operators-container">{renderOperators()}</div>
      <div className="submit-container">
        <Key
          handleKeyDown={handleKeyDown}
          keyType="submit-key"
          keyValue="="
          keyAction={callOperator}
        />
      </div>
    </div>
  );
}

KeyPad.propTypes = {
  handleKeyDown: PropTypes.func.isRequired,
  keyPadOperators: PropTypes.arrayOf(PropTypes.string).isRequired,
  keyPadNumbers: PropTypes.arrayOf(PropTypes.string).isRequired,
  updateDisplay: PropTypes.func.isRequired,
  setOperator: PropTypes.func.isRequired,
  callOperator: PropTypes.func.isRequired
};
