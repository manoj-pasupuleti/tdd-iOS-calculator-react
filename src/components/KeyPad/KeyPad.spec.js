import React from "react";
import { shallow, mount } from "enzyme";
import KeyPad from "./KeyPad";

describe("KeyPad", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <KeyPad
        handleKeyDown={jest.fn()}
        keyPadOperators={[]}
        keyPadNumbers={[]}
        updateDisplay={jest.fn()}
        setOperator={jest.fn()}
        callOperator={jest.fn()}
      />
    );
  });

  it("Should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render 4 div's", () => {
    expect(wrapper.find("div").length).toEqual(4);
  });
});

describe("Test Numbers and Operators Renderings", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <KeyPad
        handleKeyDown={jest.fn()}
        keyPadOperators={[]}
        keyPadNumbers={[]}
        updateDisplay={jest.fn()}
        setOperator={jest.fn()}
        callOperator={jest.fn()}
      />
    );
  });

  it("Test numbers rendering", () => {
    wrapper.setProps({ keyPadNumbers: ["0", "1", "2", "3"] });
    expect(wrapper.find(".numbers-container").text()).toEqual("0123");
  });

  it("Test operators rendering", () => {
    wrapper.setProps({ keyPadOperators: ["+", "-", "*", "/"] });
    expect(wrapper.find(".operators-container").text()).toEqual("+-*/");
  });
});
