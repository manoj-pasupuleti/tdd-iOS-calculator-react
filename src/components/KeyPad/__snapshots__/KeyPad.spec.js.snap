// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyPad Should render correctly 1`] = `
<div
  className="keypad-container"
>
  <div
    className="numbers-container"
  />
  <div
    className="operators-container"
  />
  <div
    className="submit-container"
  >
    <key
      handleKeyDown={[MockFunction]}
      keyAction={[MockFunction]}
      keyType="submit-key"
      keyValue="="
    />
  </div>
</div>
`;
